docs/troubleshooting-guide.md

```diff
@@ -0,0 +1,146 @@
+# Front-End Monorepo - Troubleshooting Guide
+
+At this moment in time - the distant past of October 2018 - the Zooniverse
+Front-End Monorepo is a rapidly fluctuating work in progress. New devs jumping
+in may have trouble figuring out how to even get started, so this evolving
+document was added to help with the weird errors and issues that may not yet be
+covered in the "proper" documentation.
+
+(@shaun.a.noordin 2018.10.12)
+
+NOTE: this evolving doc should be reviewed regularly and should be retired
+eventually. Any important points should be folded into the Monorepo's proper
+documentation.
+
+## Installing & Running Packages
+
+Do what the README.MD tells you:
+
+1. Clone the repo: `git clone git@github.com:zooniverse/front-end-monorepo.git`
+2. Run Lerna's bootstrap: `lerna bootstrap`
+  1. NOTE: this installs dependencies for every package, but be prepared for
+     jankiness as memory leaks might occur, files are opened for writing but
+     never closed, the OS lagging and etc. Doesn't always happen, but sometimes
+     it does.
+  2. NOTE: Despite the README saying that you should first enter into the
+     package directory before running bootstrap (`cd packages/whatever`), it
+     seems that Lerna will bootstrap _all_ packages regardless of which
+     directory you're in.
+3. Go to the package that you're interested in: `cd packages/whatever`
+4. In the package folder, you MAY want to install the dependencies manually:
+   `npm install`. This happens on certain packages where the lerna bootstrap
+   sometimes doesn't install Node modules as required. (Happens to me with
+   `app-project`.)
+5. If the package is an app, you'll probably want to run it: `npm run dev`
+6. If the package is a library, you'll probably want to build it:
+   `npm run build`
+7. If you still can't run the package, you MAY need to manually build local dependencies (see Troubleshooting below).
+
+## Troubleshooting: Installing & Running Packages
+
+**Issue 1:** During installation or bootstrap, can't install a dependency (which is a local package)
+
+- Occurs when running `lerna bootstrap` anywhere, or running `npm install` at
+  the package level.
+
+Example error message:
+```
+lerna ERR! npm install exited 1 in '@zooniverse/react-components'
+lerna ERR! npm install stderr:
+npm ERR! code E404
+npm ERR! 404 Not Found: @zooniverse/grommet-theme@2.0.0
+```
+
+Analysis:
+- This happens because Package A (react-components) requires Package B
+  (grommet-theme), and in this case, Package B needs to be **built** first
+  before Package A can use it.
+
+Solution:
+- Build Package B first, e.g.: `cd packages/grommet-theme ; npm install ; npm run build`
+- Run `lerna link` to 'register' the newly built package.
```

Comment:
I'm not sure if this is necessary since I thought the `lerna.json` config file in the root folder is what 'registers' packages. I could totally be wrong about this. @rogerhutchings ?

Reply:
In this scenario, `lerna link` was necessary.

Observations from a clean install:
- if I don't run `grommet-theme -> npm install`, then I can't run `npm run build` due to missing dependencies/the build script not working.
- if I don't run `grommet-theme -> npm run build` then I can't compile `react-components` later.
- if I don't run `lerna link`, then I can't also compile `react-components` later - presumably because one of the two steps above broke the linking. (Or maybe the initial `lerna bootstrap` didn't properly link the unbuilt grommet library to begin with?)

I'm going to try this again with the `--scope` solution provided above; hopefully if that works, it'll be a cleaner solution than `npm install/build`-ing on each package.

Reply:
Running `npm install` will break symlinks for packages. This is true whether or not you use lerna. For example, running `npm link` inside then example package folder to create the symlink, then `npm link example-package` in the app folder to tell the app to use the symlink, then if you run `npm install`, the symlink will be overrided by whatever version of the example package is specified in package.json. So, if there are symlinked packages managed by lerna, it's best to always run `lerna bootstrap` (with the scope if necessary) rather than npm install. If you're installing a new package from npm, you could run `npm install` then `lerna bootstrap` or `lerna link example-package` to redo the symlinks.